src/load_balancer/worker_hosts.rs

```diff
@@ -0,0 +1,79 @@
+// use anyhow::{Result, anyhow};
+use regex::Regex;
+
+use crate::error::LoadBalancerError;
+
+pub struct WorkerHosts {
+    pub hosts: Vec<String>
+}
+
+impl WorkerHosts {
+    pub fn new(addresses: Vec<String>) -> Result<Self, LoadBalancerError> {
+        if addresses.is_empty() {
+            return Err(LoadBalancerError::EmptyWorkerHostAddress);
+        }
+        let mut hosts = vec![];
+        for address in addresses {
+            if Self::validate(&address)? {
+                hosts.push(address);
+            }
+        }
+
+        Ok(Self { hosts })
+    }
+
+    fn validate(address: &str) -> Result<bool, LoadBalancerError> {
+        let re = Regex::new(r"^http://([a-zA-Z0-9.-]+):([0-9]{1,5})$")?;
```

Comment:
This is pretty cool. Something I was planning on doing in my implementation was using type-based validation, similarly to how we'd have an `Email` struct that validates the contents on parse 👌 So you would catch earlier errors with the struct + not have to `validate` anywhere else in the code (even though it's just once right now 😅)... Just a thought, and could be a good thing to do in an upcoming PR.